api/v1alpha1/vaultauth_types.go

```diff
@@ -25,6 +26,24 @@ type VaultAuthConfigKubernetes struct {
 	TokenExpirationSeconds int64 `json:"tokenExpirationSeconds,omitempty"`
 }
 
+// VaultAuthConfigJwt provides VaultAuth configuration options needed for authenticating to Vault.
+type VaultAuthConfigJwt struct {
+	// Role to use for authenticating to Vault.
+	Role string `json:"role"`
+	// TokenSecretKeySelector to use when referencing the secret containing the JWT token
+	// to authenticate to Vault's JWT authentication backend.
+	TokenSecretKeySelector *corev1.SecretKeySelector `json:"tokenSecretKeySelector,omitempty"`
```

Comment:
Should  I rename `TokenSecretKeySelector` to `SecretKeyRef` as in [EnvVarSource](https://pkg.go.dev/k8s.io/kubernetes@v1.27.1/pkg/apis/core#EnvVarSource)?
```
        // Selects a key of a secret in the pod's namespace.
	SecretKeyRef *SecretKeySelector